### PR TITLE
Add periodic cgroup fingerprinter

### PR DESCRIPTION
**Pull request by iverberk**
Periodically check for the availability of cgroups. Disable the exec driver if cgroups are not available.

### CODE REVIEW

File: client/fingerprint/fingerprint.go [hunk 1]
```diff
@@ -41,6 +43,7 @@ var builtinFingerprintMap = map[string]Factory{
 }
 
 // NewFingerprint is used to instantiate and return a new fingerprint
```
Comment (dadgar) [inline client/fingerprint/fingerprint.go]: Accidental new line?

File: client/fingerprint/cgroup.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		mountPointDetector: &DefaultMountPointDetector{},
+	}
+	return f
+}
+
+func (f *CGroupFingerprint) Fingerprint(cfg *client.Config, node *structs.Node) (bool, error) {
+	// Try to find a valid cgroup mount point
+	mount, err := f.mountPointDetector.MountPoint()
+	if err != nil {
+		return false, fmt.Errorf("Failed to discover cgroup mount point: %s", err)
```
Comment (dadgar) [inline client/fingerprint/cgroup.go]: Clear the attribute in the error case?

File: client/fingerprint/cgroup.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		f.clearCGroupAttributes(node)
+
+		if f.lastState == cgroupAvailable {
+			f.logger.Printf("[INFO] fingerprint.cgroups: cgroups are unavailable")
+		}
+		f.lastState = cgroupUnavailable
+		return true, nil
+	}
+
+	node.Attributes["cgroup.mountpoint"] = mount
```
Comment (dadgar) [inline client/fingerprint/cgroup.go]: Can you make the attribute `unique.cgroup.mountpoint`. The `unique.` prefixed is used on attributes to exclude them from computed node classes which is an optimization being introduced for 0.3. You can see #708 for more detail

**Comment by dadgar** [comment]
LGTM. Minor comments!

File: client/driver/exec.go [hunk 4]
```diff
@@ -60,6 +58,10 @@ func (d *ExecDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool,
 	return true, nil
 }
 
+func (d *ExecDriver) Periodic() (bool, time.Duration) {
+	return true, 15 * time.Second
```
Comment (diptanu) [inline client/driver/exec.go]: Extract 15 in a constant.

File: client/fingerprint/cgroup_linux.go [hunk 5]
```diff
@@ -0,0 +1,22 @@
+// +build linux
+
+package fingerprint
+
+import (
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+)
+
+func FindCgroupMountpointDir() (string, error) {
```
Comment (diptanu) [inline client/fingerprint/cgroup_linux.go]: Add comment

**Comment by dadgar** [comment]
Awesome! Thanks Ivo!

**Pull request merged**